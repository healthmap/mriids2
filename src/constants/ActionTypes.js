export const FETCH_EBOLA_DATA_REQUEST = "FETCH_EBOLA_DATA_REQUEST";
export const FETCH_EBOLA_DATA_SUCCESS = "FETCH_EBOLA_DATA_SUCCESS";
export const FETCH_EBOLA_DATA_FAILURE = "FETCH_EBOLA_DATA_FAILURE";

export const FETCH_EBOLA_DATA_COMBINED_REQUEST =
  "FETCH_EBOLA_DATA_COMBINED_REQUEST";
export const FETCH_EBOLA_DATA_COMBINED_SUCCESS =
  "FETCH_EBOLA_DATA_COMBINED_SUCCESS";
export const FETCH_EBOLA_DATA_COMBINED_FAILURE =
  "FETCH_EBOLA_DATA_COMBINED_FAILURE";

export const FETCH_RISK_DATA_REQUEST = "FETCH_RISK_DATA_REQUEST";
export const FETCH_RISK_DATA_SUCCESS = "FETCH_RISK_DATA_SUCCESS";
export const FETCH_RISK_DATA_FAILURE = "FETCH_RISK_DATA_FAILURE";

export const FETCH_COVID_CASE_COUNT_DATA_REQUEST =
  "FETCH_COVID_CASE_COUNT_DATA_REQUEST";
export const FETCH_COVID_CASE_COUNT_DATA_SUCCESS =
  "FETCH_COVID_CASE_COUNT_DATA_SUCCESS";
export const FETCH_COVID_CASE_COUNT_DATA_FAILURE =
  "FETCH_COVID_CASE_COUNT_DATA_FAILURE";

export const FETCH_COVID_DATA_COMBINED_REQUEST =
  "FETCH_COVID_DATA_COMBINED_REQUEST";
export const FETCH_COVID_DATA_COMBINED_SUCCESS =
  "FETCH_COVID_DATA_COMBINED_SUCCESS";
export const FETCH_COVID_DATA_COMBINED_FAILURE =
  "FETCH_COVID_DATA_COMBINED_FAILURE";

// These are used for updating the filters
export const CHANGE_COUNTRY = "CHANGE_COUNTRY";
export const CHANGE_OUTBREAK = "CHANGE_OUTBREAK";
export const CHANGE_DATE_RANGE = "CHANGE_DATE_RANGE";
export const CHANGE_VIEW = "CHANGE_VIEW";
export const CHANGE_PROJECTION = "CHANGE_PROJECTION";
