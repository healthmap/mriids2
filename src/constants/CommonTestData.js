export const reduxInitialState = {
  ebola: {
    ebolaData: {
      isFetching: 0,
      data: {},
      error: {},
    },
    ebolaDataCombined: {
      isFetching: 0,
      data: {},
      error: {},
    },
    riskData: {
      isFetching: 0,
      data: {},
      error: {},
    },
  },
  covid: {
    caseCounts: {
      isFetching: 0,
      data: [],
      error: {},
    },
    deathCounts: {
      isFetching: 0,
      data: [],
      error: {},
    },
  },
  filters: {
    country: "All",
    outbreak: "Ebola Outbreak",
    view: "snapshot",
    projection: false,
    dateRange: { from: new Date(2014, 9, 1), to: new Date(2016, 1, 20) },
  },
};

// Same as the reduxInitialState, but with the 'risk' view in the filters.
export const riskViewState = {
  ...reduxInitialState,
  filters: {
    ...reduxInitialState.filters,
    view: "risk",
  },
};
