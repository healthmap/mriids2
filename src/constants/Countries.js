export const ebolaOutbreakCountries = ["Guinea", "Liberia", "Sierra Leone"];

export const allCountries = [
  "Afghanistan",
  "Albania",
  "Algeria",
  "Andorra",
  "Angola",
  "Antigua and Barbuda",
  "Argentina",
  "Armenia",
  "Aruba",
  "Australia",
  "Austria",
  "Azerbaijan",
  "Bahamas",
  "Bahrain",
  "Bangladesh",
  "Barbados",
  "Belarus",
  "Belgium",
  "Belize",
  "Benin",
  "Bhutan",
  "Bolivia",
  "Bosnia and Herzegovina",
  "Botswana",
  "Brazil",
  "British Virgin Islands",
  "Brunei Darussalam",
  "Bulgaria",
  "Burkina Faso",
  "Burundi",
  "Cambodia",
  "Cameroon",
  "Canada",
  "Cape Verde",
  "Cayman Islands",
  "Central African Republic",
  "Chad",
  "Chile",
  "China",
  "Colombia",
  "Comoros",
  "Congo",
  "Costa Rica",
  "Cote dIvoire",
  "Croatia",
  "Cuba",
  "Cyprus",
  "Czechia",
  "Democratic Republic of the Congo",
  "Denmark",
  "Djibouti",
  "Dominica",
  "Dominican Republic",
  "Ecuador",
  "Egypt",
  "El Salvador",
  "Equatorial Guinea",
  "Eritrea",
  "Estonia",
  "Eswatini",
  "Ethiopia",
  "Fiji",
  "Finland",
  "France",
  "Gabon",
  "Gambia",
  "Georgia",
  "Germany",
  "Ghana",
  "Greece",
  "Greenland",
  "Grenada",
  "Guatemala",
  "Guinea",
  "Guinea Bissau",
  "Guyana",
  "Haiti",
  "Holy See",
  "Honduras",
  "Hungary",
  "Iceland",
  "India",
  "Indonesia",
  "Iran",
  "Iraq",
  "Ireland",
  "Israel",
  "Italy",
  "Jamaica",
  "Japan",
  "Jordan",
  "Kazakhstan",
  "Kenya",
  "Kosovo",
  "Kuwait",
  "Kyrgyzstan",
  "Laos",
  "Latvia",
  "Lebanon",
  "Lesotho",
  "Liberia",
  "Libya",
  "Liechtenstein",
  "Lithuania",
  "Luxembourg",
  "Madagascar",
  "Malawi",
  "Malaysia",
  "Maldives",
  "Mali",
  "Malta",
  "Marshall Islands",
  "Mauritania",
  "Mauritius",
  "Mexico",
  "Moldova",
  "Monaco",
  "Mongolia",
  "Montenegro",
  "Morocco",
  "Mozambique",
  "Myanmar",
  "Namibia",
  "Nepal",
  "Netherlands",
  "New Zealand",
  "Nicaragua",
  "Niger",
  "Nigeria",
  "North Macedonia",
  "Norway",
  "Oman",
  "Pakistan",
  "Panama",
  "Papua New Guinea",
  "Paraguay",
  "Peru",
  "Philippines",
  "Poland",
  "Portugal",
  "Puerto Rico",
  "Qatar",
  "Romania",
  "Russia",
  "Rwanda",
  "Saint Kitts and Nevis",
  "Saint Lucia",
  "Saint Vincent and the Grenadines",
  "Samoa",
  "San Marino",
  "Sao Tome and Principe",
  "Saudi Arabia",
  "Senegal",
  "Serbia",
  "Seychelles",
  "Sierra Leone",
  "Singapore",
  "Slovakia",
  "Slovenia",
  "Solomon Islands",
  "Somalia",
  "South Africa",
  "South Korea",
  "South Sudan",
  "Spain",
  "Sri Lanka",
  "Sudan",
  "Suriname",
  "Sweden",
  "Switzerland",
  "Syria",
  "Taiwan",
  "Tajikistan",
  "Thailand",
  "Timor Leste",
  "Togo",
  "Trinidad and Tobago",
  "Tunisia",
  "Turkey",
  "Uganda",
  "Ukraine",
  "United Arab Emirates",
  "United Kingdom",
  "United Republic of Tanzania",
  "United States of America",
  "Uruguay",
  "Uzbekistan",
  "Vanuatu",
  "Venezuela",
  "Vietnam",
  "Yemen",
  "Zambia",
  "Zimbabwe",
];
