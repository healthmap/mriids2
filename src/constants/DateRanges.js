export const ebolaInitialDateRange = {
  from: new Date(2014, 9, 1),
  to: new Date(2016, 1, 20),
};
