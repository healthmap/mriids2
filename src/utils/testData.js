// This file has test data for the unit tests.

export const testGuineaData = {
  Guinea: {
    "2014-10-06": {
      projections: {
        oneWeek: 179,
        twoWeeks: 210.5,
        threeWeeks: 251,
        fourWeeks: 296,
      },
      value: "126",
    },
    "2014-10-13": {
      projections: {
        oneWeek: 153,
        twoWeeks: 163,
        threeWeeks: 175,
        fourWeeks: 190,
      },
      value: "114",
    },
  },
};

export const testGuineaDataOutOfDateRange = {
  Guinea: {
    "2013-10-06": {
      projections: {
        oneWeek: 179,
        twoWeeks: 210.5,
        threeWeeks: 251,
        fourWeeks: 296,
      },
      value: "126",
    },
    "2014-10-13": {
      projections: {
        oneWeek: 153,
        twoWeeks: 163,
        threeWeeks: 175,
        fourWeeks: 190,
      },
      value: "114",
    },
  },
};

export const allCountriesEbolaData = {
  Guinea: {
    "2014-10-06": {
      projections: {
        oneWeek: 179,
        twoWeeks: 210.5,
        threeWeeks: 251,
        fourWeeks: 296,
      },
      value: "126",
    },
  },
  Liberia: {
    "2014-10-06": {
      projections: {
        oneWeek: 179,
        twoWeeks: 210.5,
        threeWeeks: 251,
        fourWeeks: 296,
      },
      value: "126",
    },
  },
  "Sierra Leone": {
    "2014-10-06": {
      projections: {
        oneWeek: 179,
        twoWeeks: 210.5,
        threeWeeks: 251,
        fourWeeks: 296,
      },
      value: "126",
    },
  },
};

export const allCountriesEbolaDataLiberiaOutOfDateRange = {
  Guinea: {
    "2014-10-06": {
      projections: {
        oneWeek: 179,
        twoWeeks: 210.5,
        threeWeeks: 251,
        fourWeeks: 296,
      },
      value: "126",
    },
  },
  Liberia: {
    "2013-10-06": {
      projections: {
        oneWeek: 179,
        twoWeeks: 210.5,
        threeWeeks: 251,
        fourWeeks: 296,
      },
      value: "126",
    },
  },
  "Sierra Leone": {
    "2014-10-06": {
      projections: {
        oneWeek: 179,
        twoWeeks: 210.5,
        threeWeeks: 251,
        fourWeeks: 296,
      },
      value: "126",
    },
  },
};

export const testRawEbolaData = {
  Guinea: {
    "2014-10-06": {
      projections: {
        oneWeek: 179,
        twoWeeks: 210.5,
        threeWeeks: 251,
        fourWeeks: 296,
      },
      value: "2452",
    },
  },
  Liberia: {
    "2014-10-06": {
      projections: {
        oneWeek: 179,
        twoWeeks: 210.5,
        threeWeeks: 251,
        fourWeeks: 296,
      },
      value: "6738",
    },
  },
  "Sierra Leone": {
    "2014-10-06": {
      projections: {
        oneWeek: 179,
        twoWeeks: 210.5,
        threeWeeks: 251,
        fourWeeks: 296,
      },
      value: "11387",
    },
  },
};

export const ebolaFillColorDictionary = {
  Guinea: "#F5BCA7",
  Liberia: "#EB7C77",
  "Sierra Leone": "#E23D4A",
};

export const testGuineaFiltersState = {
  country: "Guinea",
  outbreak: "Ebola Outbreak",
  view: "snapshot",
  projection: false,
  dateRange: { from: new Date(2014, 9, 1), to: new Date(2016, 1, 20) },
};

export const testEbolaDataCombined = [
  {
    week_starting: "2014-10-05",
    projection_from: "2014-10-06",
    Liberia: "290",
    "Sierra Leone": "480",
    Guinea: "126",
    aggregated: "896",
    "y1.aggregated": "1336",
    "y1.Guinea": "179",
    "y1.Liberia": "560",
    "y1.Sierra Leone": "592",
    "y2.aggregated": "1598",
    "y2.Guinea": "210.5",
    "y2.Liberia": "582",
    "y2.Sierra Leone": "745",
    "y3.aggregated": "1946",
    "y3.Guinea": "251",
    "y3.Liberia": "639",
    "y3.Sierra Leone": "920.5",
    "y4.aggregated": "2392",
    "y4.Guinea": "296",
    "y4.Liberia": "727",
    "y4.Sierra Leone": "1141.5",
    "ymin1.aggregated": "1203",
    "ymin1.Guinea": "144",
    "ymin1.Liberia": "281",
    "ymin1.Sierra Leone": "526.975",
    "ymin2.aggregated": "1457.975",
    "ymin2.Guinea": "171",
    "ymin2.Liberia": "383",
    "ymin2.Sierra Leone": "648.975",
    "ymin3.aggregated": "1740.9",
    "ymin3.Guinea": "204",
    "ymin3.Liberia": "516.975",
    "ymin3.Sierra Leone": "780",
    "ymin4.aggregated": "2090.875",
    "ymin4.Guinea": "240",
    "ymin4.Liberia": "631.95",
    "ymin4.Sierra Leone": "943.975",
    "ymax1.aggregated": "1475.05",
    "ymax1.Guinea": "247",
    "ymax1.Liberia": "626",
    "ymax1.Sierra Leone": "887.025",
    "ymax2.aggregated": "1773.175",
    "ymax2.Guinea": "340.025",
    "ymax2.Liberia": "655.025",
    "ymax2.Sierra Leone": "1230.025",
    "ymax3.aggregated": "2199.025",
    "ymax3.Guinea": "481",
    "ymax3.Liberia": "732.025",
    "ymax3.Sierra Leone": "1732",
    "ymax4.aggregated": "2798.15",
    "ymax4.Guinea": "682.025",
    "ymax4.Liberia": "891.125",
    "ymax4.Sierra Leone": "2452.075",
  },
  {
    week_starting: "2014-10-12",
    projection_from: "2014-10-13",
    Liberia: "320",
    "Sierra Leone": "425",
    Guinea: "114",
    aggregated: "859",
    "y1.aggregated": "950",
    "y1.Guinea": "153",
    "y1.Liberia": "229",
    "y1.Sierra Leone": "566",
    "y2.aggregated": "987.5",
    "y2.Guinea": "163",
    "y2.Liberia": "192",
    "y2.Sierra Leone": "630",
    "y3.aggregated": "1047.5",
    "y3.Guinea": "175",
    "y3.Liberia": "161",
    "y3.Sierra Leone": "709",
    "y4.aggregated": "1127",
    "y4.Guinea": "190",
    "y4.Liberia": "138",
    "y4.Sierra Leone": "791",
    "ymin1.aggregated": "882",
    "ymin1.Guinea": "118",
    "ymin1.Liberia": "189",
    "ymin1.Sierra Leone": "493",
    "ymin2.aggregated": "914.95",
    "ymin2.Guinea": "126",
    "ymin2.Liberia": "153.975",
    "ymin2.Sierra Leone": "527.975",
    "ymin3.aggregated": "966",
    "ymin3.Guinea": "131",
    "ymin3.Liberia": "123.975",
    "ymin3.Sierra Leone": "568",
    "ymin4.aggregated": "1026.925",
    "ymin4.Guinea": "134.975",
    "ymin4.Liberia": "101.975",
    "ymin4.Sierra Leone": "607.975",
    "ymax1.aggregated": "1017.025",
    "ymax1.Guinea": "191",
    "ymax1.Liberia": "270",
    "ymax1.Sierra Leone": "644.05",
    "ymax2.aggregated": "1061.025",
    "ymax2.Guinea": "206",
    "ymax2.Liberia": "235",
    "ymax2.Sierra Leone": "747",
    "ymax3.aggregated": "1137.025",
    "ymax3.Guinea": "232",
    "ymax3.Liberia": "205",
    "ymax3.Sierra Leone": "869",
    "ymax4.aggregated": "1238",
    "ymax4.Guinea": "261",
    "ymax4.Liberia": "181.025",
    "ymax4.Sierra Leone": "1024",
  },
];

export const testEbolaDataCombinedOutOfDateRange = [
  {
    week_starting: "2014-10-05",
    projection_from: "2013-10-06",
    Liberia: "290",
    "Sierra Leone": "480",
    Guinea: "126",
    aggregated: "896",
    "y1.aggregated": "1336",
    "y1.Guinea": "179",
    "y1.Liberia": "560",
    "y1.Sierra Leone": "592",
    "y2.aggregated": "1598",
    "y2.Guinea": "210.5",
    "y2.Liberia": "582",
    "y2.Sierra Leone": "745",
    "y3.aggregated": "1946",
    "y3.Guinea": "251",
    "y3.Liberia": "639",
    "y3.Sierra Leone": "920.5",
    "y4.aggregated": "2392",
    "y4.Guinea": "296",
    "y4.Liberia": "727",
    "y4.Sierra Leone": "1141.5",
    "ymin1.aggregated": "1203",
    "ymin1.Guinea": "144",
    "ymin1.Liberia": "281",
    "ymin1.Sierra Leone": "526.975",
    "ymin2.aggregated": "1457.975",
    "ymin2.Guinea": "171",
    "ymin2.Liberia": "383",
    "ymin2.Sierra Leone": "648.975",
    "ymin3.aggregated": "1740.9",
    "ymin3.Guinea": "204",
    "ymin3.Liberia": "516.975",
    "ymin3.Sierra Leone": "780",
    "ymin4.aggregated": "2090.875",
    "ymin4.Guinea": "240",
    "ymin4.Liberia": "631.95",
    "ymin4.Sierra Leone": "943.975",
    "ymax1.aggregated": "1475.05",
    "ymax1.Guinea": "247",
    "ymax1.Liberia": "626",
    "ymax1.Sierra Leone": "887.025",
    "ymax2.aggregated": "1773.175",
    "ymax2.Guinea": "340.025",
    "ymax2.Liberia": "655.025",
    "ymax2.Sierra Leone": "1230.025",
    "ymax3.aggregated": "2199.025",
    "ymax3.Guinea": "481",
    "ymax3.Liberia": "732.025",
    "ymax3.Sierra Leone": "1732",
    "ymax4.aggregated": "2798.15",
    "ymax4.Guinea": "682.025",
    "ymax4.Liberia": "891.125",
    "ymax4.Sierra Leone": "2452.075",
  },
  {
    week_starting: "2014-10-12",
    projection_from: "2014-10-13",
    Liberia: "320",
    "Sierra Leone": "425",
    Guinea: "114",
    aggregated: "859",
    "y1.aggregated": "950",
    "y1.Guinea": "153",
    "y1.Liberia": "229",
    "y1.Sierra Leone": "566",
    "y2.aggregated": "987.5",
    "y2.Guinea": "163",
    "y2.Liberia": "192",
    "y2.Sierra Leone": "630",
    "y3.aggregated": "1047.5",
    "y3.Guinea": "175",
    "y3.Liberia": "161",
    "y3.Sierra Leone": "709",
    "y4.aggregated": "1127",
    "y4.Guinea": "190",
    "y4.Liberia": "138",
    "y4.Sierra Leone": "791",
    "ymin1.aggregated": "882",
    "ymin1.Guinea": "118",
    "ymin1.Liberia": "189",
    "ymin1.Sierra Leone": "493",
    "ymin2.aggregated": "914.95",
    "ymin2.Guinea": "126",
    "ymin2.Liberia": "153.975",
    "ymin2.Sierra Leone": "527.975",
    "ymin3.aggregated": "966",
    "ymin3.Guinea": "131",
    "ymin3.Liberia": "123.975",
    "ymin3.Sierra Leone": "568",
    "ymin4.aggregated": "1026.925",
    "ymin4.Guinea": "134.975",
    "ymin4.Liberia": "101.975",
    "ymin4.Sierra Leone": "607.975",
    "ymax1.aggregated": "1017.025",
    "ymax1.Guinea": "191",
    "ymax1.Liberia": "270",
    "ymax1.Sierra Leone": "644.05",
    "ymax2.aggregated": "1061.025",
    "ymax2.Guinea": "206",
    "ymax2.Liberia": "235",
    "ymax2.Sierra Leone": "747",
    "ymax3.aggregated": "1137.025",
    "ymax3.Guinea": "232",
    "ymax3.Liberia": "205",
    "ymax3.Sierra Leone": "869",
    "ymax4.aggregated": "1238",
    "ymax4.Guinea": "261",
    "ymax4.Liberia": "181.025",
    "ymax4.Sierra Leone": "1024",
  },
];

export const testEbolaDataCombinedLastWeekOutOfDateRange = [
  {
    week_starting: "2014-10-05",
    projection_from: "2014-10-06",
    Liberia: "290",
    "Sierra Leone": "480",
    Guinea: "126",
    aggregated: "896",
    "y1.aggregated": "1336",
    "y1.Guinea": "179",
    "y1.Liberia": "560",
    "y1.Sierra Leone": "592",
    "y2.aggregated": "1598",
    "y2.Guinea": "210.5",
    "y2.Liberia": "582",
    "y2.Sierra Leone": "745",
    "y3.aggregated": "1946",
    "y3.Guinea": "251",
    "y3.Liberia": "639",
    "y3.Sierra Leone": "920.5",
    "y4.aggregated": "2392",
    "y4.Guinea": "296",
    "y4.Liberia": "727",
    "y4.Sierra Leone": "1141.5",
    "ymin1.aggregated": "1203",
    "ymin1.Guinea": "144",
    "ymin1.Liberia": "281",
    "ymin1.Sierra Leone": "526.975",
    "ymin2.aggregated": "1457.975",
    "ymin2.Guinea": "171",
    "ymin2.Liberia": "383",
    "ymin2.Sierra Leone": "648.975",
    "ymin3.aggregated": "1740.9",
    "ymin3.Guinea": "204",
    "ymin3.Liberia": "516.975",
    "ymin3.Sierra Leone": "780",
    "ymin4.aggregated": "2090.875",
    "ymin4.Guinea": "240",
    "ymin4.Liberia": "631.95",
    "ymin4.Sierra Leone": "943.975",
    "ymax1.aggregated": "1475.05",
    "ymax1.Guinea": "247",
    "ymax1.Liberia": "626",
    "ymax1.Sierra Leone": "887.025",
    "ymax2.aggregated": "1773.175",
    "ymax2.Guinea": "340.025",
    "ymax2.Liberia": "655.025",
    "ymax2.Sierra Leone": "1230.025",
    "ymax3.aggregated": "2199.025",
    "ymax3.Guinea": "481",
    "ymax3.Liberia": "732.025",
    "ymax3.Sierra Leone": "1732",
    "ymax4.aggregated": "2798.15",
    "ymax4.Guinea": "682.025",
    "ymax4.Liberia": "891.125",
    "ymax4.Sierra Leone": "2452.075",
  },
  {
    week_starting: "2014-10-12",
    projection_from: "2020-10-13",
    Liberia: "320",
    "Sierra Leone": "425",
    Guinea: "114",
    aggregated: "859",
    "y1.aggregated": "950",
    "y1.Guinea": "153",
    "y1.Liberia": "229",
    "y1.Sierra Leone": "566",
    "y2.aggregated": "987.5",
    "y2.Guinea": "163",
    "y2.Liberia": "192",
    "y2.Sierra Leone": "630",
    "y3.aggregated": "1047.5",
    "y3.Guinea": "175",
    "y3.Liberia": "161",
    "y3.Sierra Leone": "709",
    "y4.aggregated": "1127",
    "y4.Guinea": "190",
    "y4.Liberia": "138",
    "y4.Sierra Leone": "791",
    "ymin1.aggregated": "882",
    "ymin1.Guinea": "118",
    "ymin1.Liberia": "189",
    "ymin1.Sierra Leone": "493",
    "ymin2.aggregated": "914.95",
    "ymin2.Guinea": "126",
    "ymin2.Liberia": "153.975",
    "ymin2.Sierra Leone": "527.975",
    "ymin3.aggregated": "966",
    "ymin3.Guinea": "131",
    "ymin3.Liberia": "123.975",
    "ymin3.Sierra Leone": "568",
    "ymin4.aggregated": "1026.925",
    "ymin4.Guinea": "134.975",
    "ymin4.Liberia": "101.975",
    "ymin4.Sierra Leone": "607.975",
    "ymax1.aggregated": "1017.025",
    "ymax1.Guinea": "191",
    "ymax1.Liberia": "270",
    "ymax1.Sierra Leone": "644.05",
    "ymax2.aggregated": "1061.025",
    "ymax2.Guinea": "206",
    "ymax2.Liberia": "235",
    "ymax2.Sierra Leone": "747",
    "ymax3.aggregated": "1137.025",
    "ymax3.Guinea": "232",
    "ymax3.Liberia": "205",
    "ymax3.Sierra Leone": "869",
    "ymax4.aggregated": "1238",
    "ymax4.Guinea": "261",
    "ymax4.Liberia": "181.025",
    "ymax4.Sierra Leone": "1024",
  },
];

export const testCovidData = [
  {
    country: "Canada",
    province: [
      "alberta",
      "british columbia",
      "diamond princess",
      "grand princess",
      "manitoba",
      "new brunswick",
      "newfoundland and labrador",
      "northwest territories",
      "nova scotia",
      "nunavut",
      "ontario",
      "prince edward island",
      "quebec",
      "repatriated travellers",
      "saskatchewan",
      "yukon",
    ],
    timeline: {
      cases: {
        "11/11/20": 280465,
        "11/12/20": 285939,
      },
      deaths: {
        "11/11/20": 10748,
        "11/12/20": 10828,
      },
      recovered: {
        "11/11/20": 0,
        "11/12/20": 0,
      },
    },
  },
  {
    country: "Honduras",
    province: ["mainland"],
    timeline: {
      cases: {
        "11/11/20": 101169,
        "11/12/20": 101468,
      },
      deaths: {
        "11/11/20": 2797,
        "11/12/20": 2804,
      },
      recovered: {
        "11/11/20": 43857,
        "11/12/20": 44068,
      },
    },
  },
  {
    message: "Country not found or doesn't have any historical data",
  },
];

export const testParsedCovidData = [
  {
    countryName: "Afghanistan",
    cases: { "10/28/20": 30000, "11/28/20": 40000, "11/29/20": 46215 },
    deaths: { "11/29/20": 1763 },
  },
  {
    countryName: "Zimbabwe",
    cases: { "10/28/20": 30000, "11/28/20": 40000, "11/29/20": 46215 },
    deaths: { "11/29/20": 1763 },
  },
];

export const testObjectDateKeys = {
  "1/22/20": 0,
  "1/23/20": 0,
  "1/24/20": 0,
  "1/25/20": 0,
  "1/26/20": 0,
  "1/27/20": 0,
  "1/28/20": 0,
  "1/29/20": 0,
  "1/30/20": 0,
  "1/31/20": 0,
  "2/1/20": 0,
  "2/2/20": 0,
  "2/3/20": 0,
  "2/4/20": 0,
  "2/5/20": 0,
  "2/6/20": 0,
  "2/7/20": 0,
  "2/8/20": 0,
  "2/9/20": 0,
  "2/10/20": 0,
  "2/11/20": 0,
  "2/12/20": 0,
};
