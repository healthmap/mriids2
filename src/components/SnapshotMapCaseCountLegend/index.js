import React from "react";
import { connect } from "react-redux";
import MapLegendLevel from "../MapLegendLevel";
import {
  getEbolaScale,
  getCovidScale,
  getSnapshotColor,
  getSnapshotProjectionsColor,
} from "../../utils/snapshotMapHelpers";
import { getCountriesEbolaCaseCounts } from "../../utils/ebolaDataHelpers";
import { getCountriesCovidCaseCounts } from "../../utils/covidDataHelpers";
import {
  MapLegendWrapperSnapshot,
  MapLegendItemsWrapper,
} from "../styled-components/MapLegendWrappers";
import { BlockDropshadow } from "../styled-components/Block";

const SnapshotMapCaseCountLegend = ({ ebolaData, covidData, filters }) => {
  // Determines whether the ebola outbreak is selected.
  const ebolaOutbreakSelected = filters.outbreak === "Ebola Outbreak";

  // Getting the number of case counts for each country.
  // We need this to get the max value for the legend.
  const countriesCaseCounts = ebolaOutbreakSelected
    ? getCountriesEbolaCaseCounts(ebolaData, filters)
    : getCountriesCovidCaseCounts(covidData, filters);

  const legendHeader = filters.projection
    ? "Total outbreak projections"
    : "Case counts";

  const renderLegendLevels = () => {
    const scale = ebolaOutbreakSelected
      ? getEbolaScale(countriesCaseCounts)
      : getCovidScale(countriesCaseCounts);
    // We want to render 10 levels for the legend.
    const numberOfLevels = 9;
    const levels = [];
    for (let i = 0; i <= numberOfLevels; i++) {
      let value = i / numberOfLevels;
      // If the projections are enabled, we want to use the projections colors.
      // Else, use the regular snapshot colors.
      const color = filters.projection
        ? getSnapshotProjectionsColor(value)
        : getSnapshotColor(value);
      levels.push(
        <MapLegendLevel
          key={`legend-level-${i}`}
          color={color}
          value={Math.round(value * scale).toLocaleString()}
        />
      );
    }
    // We reverse the order of the elements in the 'levels' array because we want to highest level at the top.
    return levels.reverse();
  };

  return (
    <MapLegendWrapperSnapshot>
      <BlockDropshadow>
        <h3>{legendHeader}</h3>
        <MapLegendItemsWrapper>{renderLegendLevels()}</MapLegendItemsWrapper>
      </BlockDropshadow>
    </MapLegendWrapperSnapshot>
  );
};

const mapStateToProps = (state) => ({
  ebolaData: state.ebola.ebolaData.data,
  covidData: state.covid.covidData.data,
  filters: state.filters,
});

export default connect(mapStateToProps)(SnapshotMapCaseCountLegend);
