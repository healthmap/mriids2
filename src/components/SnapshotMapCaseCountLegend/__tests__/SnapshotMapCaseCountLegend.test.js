import React from "react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { reduxInitialState } from "../../../constants/CommonTestData";
import { ebolaCaseCountsDictionary } from "../../../utils/testData/ebolaTestData";
import SnapshotMapCaseCountLegend from "../index";

const mockStore = configureStore([thunk]);

describe("Tests for the connected SnapshotMapCaseCountLegend component with reduxInitialState", () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore(reduxInitialState);

    component = renderer.create(
      <Provider store={store}>
        <SnapshotMapCaseCountLegend
          countryDiseaseCounts={ebolaCaseCountsDictionary}
        />
      </Provider>
    );
  });
  test("should render with given state from Redux store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
