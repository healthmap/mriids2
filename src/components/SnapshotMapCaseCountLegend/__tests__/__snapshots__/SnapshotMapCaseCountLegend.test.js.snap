// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the connected SnapshotMapCaseCountLegend component with reduxInitialState should render with given state from Redux store 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 2px;
}

.c4 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  width: 2em;
  height: 2em;
  margin-right: 0.5em;
}

.c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  color: #666;
}

.c0 {
  position: absolute;
  left: 2rem;
  bottom: 3.2rem;
  width: 9rem;
}

.c0 > div {
  width: 100%;
}

.c2 {
  height: calc(100% - 7rem);
  overflow: auto;
}

.c1 {
  display: inline-block;
  padding: 1rem;
  box-shadow: 1px 1px 6px 0px rgba(0,0,0,0.1);
  border-radius: 0.5rem;
  background-color: #fff;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h3>
      Case counts
    </h3>
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <div
          className="c4"
          style={
            Object {
              "backgroundColor": "#E23D4A",
            }
          }
        />
        <div
          className="c5"
        >
          20
        </div>
      </div>
      <div
        className="c3"
      >
        <div
          className="c4"
          style={
            Object {
              "backgroundColor": "#E23D4A",
            }
          }
        />
        <div
          className="c5"
        >
          18
        </div>
      </div>
      <div
        className="c3"
      >
        <div
          className="c4"
          style={
            Object {
              "backgroundColor": "#E55259",
            }
          }
        />
        <div
          className="c5"
        >
          16
        </div>
      </div>
      <div
        className="c3"
      >
        <div
          className="c4"
          style={
            Object {
              "backgroundColor": "#E86769",
            }
          }
        />
        <div
          className="c5"
        >
          13
        </div>
      </div>
      <div
        className="c3"
      >
        <div
          className="c4"
          style={
            Object {
              "backgroundColor": "#EB7C77",
            }
          }
        />
        <div
          className="c5"
        >
          11
        </div>
      </div>
      <div
        className="c3"
      >
        <div
          className="c4"
          style={
            Object {
              "backgroundColor": "#EE9187",
            }
          }
        />
        <div
          className="c5"
        >
          9
        </div>
      </div>
      <div
        className="c3"
      >
        <div
          className="c4"
          style={
            Object {
              "backgroundColor": "#F1A697",
            }
          }
        />
        <div
          className="c5"
        >
          7
        </div>
      </div>
      <div
        className="c3"
      >
        <div
          className="c4"
          style={
            Object {
              "backgroundColor": "#F5BCA7",
            }
          }
        />
        <div
          className="c5"
        >
          4
        </div>
      </div>
      <div
        className="c3"
      >
        <div
          className="c4"
          style={
            Object {
              "backgroundColor": "#F8D1B6",
            }
          }
        />
        <div
          className="c5"
        >
          2
        </div>
      </div>
      <div
        className="c3"
      >
        <div
          className="c4"
          style={
            Object {
              "backgroundColor": "#FDF1DD",
            }
          }
        />
        <div
          className="c5"
        >
          0
        </div>
      </div>
    </div>
  </div>
</div>
`;
